### ISSUE TITLE
NVDA does not read the contents of some text fields in Firefox

### ISSUE THREAD

**Issue by parhamdoustdar** [nvaccess/nvda#5861]
This has happened to me on some other websites. However, I forget them now. Seeing this on a website very recently, I thought I'd write it down before I forget.

Steps to reproduce:
1. Go to https://www.hackerrank.com/signup?utm_medium=header&utm_source=hr-homepage
2. Pressing `e` will take you to the first text field, namely `First name`. Write something in that text field.
3. Press the keystroke to read the current line (in the laptop layout, it's `insert+L`).

Expected result: your name should be spoken.

Actual result: only the first letter is spoken.

**Comment by PratikP1** [comment]
I am unable to replicate this with the latest version of Firefox and the latest NVDA snapshot. The full field is read. I have, however, seen this behavior.

**Comment by bhavyashah** [comment]
Hi,
Having not been much active on the NVDA issue tracker recently, I too
missed filing this bug.
Another very prominent example to consistently replicate this
behaviour is the To: field in GMail Basic HTML View Compose Mail page
with latest stable versions of NVDA and Firefox respectively.
Thanks.

On 4/6/16, PratikP1 <notifications@github.com> wrote:
> I am unable to replicate this with the latest version of Firefox and the
> latest NVDA snapshot. The full field is read. I have, however, seen this
> behavior.
>
>
> ---
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/5861#issuecomment-206439470


-- 
Warm Regards
Bhavya Shah
Using NVDA (Non Visual Desktop Access) free and open source screen
reader for Microsoft Windows
To download a copy of the free screen reader NVDA, please visit
http://www.nvaccess.org/
Using Google Talkback on Motorolla G second generation Lollipop 5.0.2
Reach me through the following means:
Mobile: +91 7506221750
E-mail id: bhavya.shah125@gmail.com
Skype id : bhavya.09

**Comment by jcsteh** [comment]
I can't reproduce this. For me, the entire line is read.

What version of Firefox are you running?

**Comment by jcsteh** [comment]
I can't reproduce it in the To: field of Gmail basic HTML either. Again, 
the entire line is read.

I'm wondering whether your Firefox window is extremely small or the 
like. Please ensure it's maximised (alt+space then x) and also that you 
are using the default zoom level (press control+0).

**Comment by parhamdoustdar** [comment]
I pressed `ctrl+0` and also `alt+space, x`. However, the error still persists.

Firefox version: 45.0.1

NVDA version: master-13281,1aae7484

**Comment by MattTreichelYeah** [comment]
I don't have a good example to add because it's behind a secure page, but I thought I'd say I've seen similar behavior. NVDA only in Firefox only reads the first letter of a tabbed to div (has tabindex=0), but with varying degrees of randomness reads it correctly under identical tabbing behaviour (even alternating pronunciation between subsequent tabbing on/off), and reads correctly in following identically structured divs.

This is in Firefox 46.0.1 and NVDA 2016.2.

**Comment by ehollig** [comment]
@parhamdoustdar or @bhavyashah are you still able to reproduce this?

**Comment by bhavyashah** [comment]
@ehollig Yes, I can still reproduce this issue with NVDA 2018.3 Beta 2, Firefox 61.0.2 and Windows 8.1. At the risk of redundantly reiterating, but just to ensure that we are all absolutely on the same page, below are the steps to reproduce this issue with the original test case:
1. Open https://www.hackerrank.com/signup?utm_medium=header&utm_source=hr-homepage on Firefox.
2. Navigate to the first edit box on this webpage labelled “First and Last Name “.
3. Enter this edit box, either by pressing Enter or Spacebar while focusing it, by Tabbing into it, by activating focus mode while on it or through an end number of other methods.
4. Now that you are placed inside this edit box, type several characters of text so that they get inputted in the field. E.g. testing
5. Issue the read current line command (NVDA+L for the laptop keyboard layout) to access the contents of this edit field that you just typed in.
### Expected Result: ###
NVDA will read “testing”.
### Actual Result: ###
NVDA says “t”.

**Comment by Adriani90** [comment]
@jcsteh Ok I found a way how to reproduce this reliably:
1. Open NVDA and the webpage from issue description
2. in Browse mode, Press e to go to the first edit field
3. Press down arrow, then up arrow
4. Press enter and type your name
5. Press nvda+l
Actual: NVDA reports only the first letter.
Note: When pressing down and up arrow in browse mode (step 3), a notification is shown which however is only reported in focus mode. In browse mode you hear just "notification". Now the notification event  seems to change NVDA's behavior. To see what I mean do following:
1. Open the website
2. In browse mode, press e and then enter
3. Type your name and press nvda+l
Actual: the full text is spoken

Conclusion: As long as the notification is not displayed, the full name is spoken. after that only the first letter is spoken.

**Comment by Adriani90** [comment]
the same behavior is in focus mode if you use tab and shift+tab instead of down arrow and up arrow.

**Comment by lukaszgo1** [comment]
#9001 maybe related

**Comment by lukaszgo1** [comment]
Still reproducible with the latest Nightly as of today and latest NVDA Alpha. This also reproduces with JAWS so it looks like a Firefox issue. cc @jcsteh @MarcoZehe

**Comment by jcsteh** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1637779

**Comment by jcsteh** [comment]
I just tested this and I can't reproduce it any more, even with Cache the World disabled. Can anyone still reproduce it?

**Comment by seanbudd** [comment]
Closing as fixed/stale/cannot reproduce

If this is still an issue, let us know so we can re-open it

**Comment by bhavyashah** [comment]
Just for completeness, I'll add that I cannot reproduce this issue with the To field in GMail Basic HTML with latest stable versions of Firefox and NVDA.
